solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -2605,95 +2636,152 @@ public SolrMetricsContext getSolrMetricsContext() {
+              } catch (Exception ignore) {
+                // replacement for nullNumber
+              }
+            }));
+
+    toClose.add(
+        solrMetricsContext.observableLongGauge(
+            "solr_searcher_max_doc",
+            "Max doc (including deletions)",
+            obs -> {
+              try {
+                obs.record(reader.maxDoc(), baseAttributes);
+              } catch (Exception ignore) {
+              }
+            }));
+
+    toClose.add(
+        solrMetricsContext.observableLongGauge(
+            "solr_searcher_deleted_docs",
+            "Number of deleted docs",
+            obs -> {
+              try {
+                obs.record(reader.maxDoc() - reader.numDocs(), baseAttributes);
+              } catch (Exception ignore) {
+              }
+            }));
+    // indexVersion (numeric)
+    toClose.add(
+        solrMetricsContext.observableLongGauge(
+            "solr_searcher_index_version",
+            "Lucene index version",
+            obs -> {
+              try {
+                obs.record(reader.getVersion(), baseAttributes);
+              } catch (Exception ignore) {
+              }
+            }));
     // size of the currently opened commit
-    parentContext.gauge(
-        () -> {
-          try {
-            Collection<String> files = reader.getIndexCommit().getFileNames();
-            long total = 0;
-            for (String file : files) {
-              total += DirectoryFactory.sizeOf(reader.directory(), file);
-            }
-            return total;
-          } catch (Exception e) {
-            return parentContext.nullNumber();
-          }
-        },
-        true,
-        "indexCommitSize",
-        Category.SEARCHER.toString(),
-        scope);
+    toClose.add(
+        solrMetricsContext.observableLongGauge(
+            "solr_searcher_index_commit_size_bytes",
+            "Size of the current index commit (bytes)",
+            obs -> {
+              try {
+                long total = 0L;
+                for (String file : reader.getIndexCommit().getFileNames()) {
+                  total += DirectoryFactory.sizeOf(reader.directory(), file);
+                }
+                obs.record(total, baseAttributes);
+              } catch (Exception e) {
+                // skip recording if unavailable (no nullNumber in OTel)
+              }
+            }));
+
+    var cacheBaseAttribute =
+        attributes.toBuilder().put(CATEGORY_ATTR, Category.CACHE.toString()).build();
     // statsCache metrics
-    parentContext.gauge(
-        new MetricsMap(
-            map -> {
-              statsCache.getCacheMetrics().getSnapshot(map::putNoEx);
-              map.put("statsCacheImpl", statsCache.getClass().getSimpleName());
-            }),
-        true,
-        "statsCache",
-        Category.CACHE.toString(),
-        scope);
+    toClose.add(
+        solrMetricsContext.observableLongGauge(
+            "solr_searcher_stats_cache",
```

Comment:
The `StatsCache` class should have been named `TermStatsCache`.  Its current name is *way* too generic looking!  _What_ stats?!  (CC @sigram )
While changing the name of that is definitely out of scope, what **is** in scope here is thinking of clear metric names we expose.  I propose that its metric name incorporate "termstats" instead of just "stats".

Reply:
Also WDYT of moving this initialization to StatsCache (perhaps a static method) so that SolrIndexSearcher can be less distracted by details of that cache.  We do similarly for CaffeineCache.  SolrIndexSearcher is a huge class.